Blockzilla/Lib/DomainCompletion/DomainCompletion.swift

```diff
@@ -33,32 +33,36 @@ protocol CustomAutocompleteSource: AutocompleteSource {
 }
 
 class CustomCompletionSource: CustomAutocompleteSource {
-
+    private lazy var regex = try! NSRegularExpression(pattern: "^(\\s+)?(?:https?:\\/\\/)?(?:www\\.)?", options: [.caseInsensitive])
     var enabled: Bool { return Settings.getToggle(.enableCustomDomainAutocomplete) }
 
     func getSuggestions() -> AutoCompleteSuggestions {
         return Settings.getCustomDomainSetting()
     }
 
     func add(suggestion: String) -> CustomCompletionResult {
-        guard !suggestion.isEmpty else { return .error(.invalidUrl) }
+        let sanitizedSuggestion = regex.stringByReplacingMatches(in: suggestion, options: [], range: NSMakeRange(0, suggestion.count), withTemplate: "")
```

Comment:
So this regex is the perfect thing that could be unit tested. :)

Reply:
Definitely agree. We gotta get a Unit testing project in here asap!